pulpcore/cli/common/openapi.py

```diff
@@ -142,11 +142,23 @@ def extract_params(
         )
         result: Dict[str, Any] = {}
         for name in list(params.keys()):
-            if name in param_spec:
-                param_spec.pop(name)
-                result[name] = params.pop(name)
+            if name in param_specs:
+                param = params.pop(name)
+                param_spec = param_specs.pop(name)
+                param_schema = param_spec.get("schema")
+                if param_schema:
+                    param_type = param_schema.get("type", "string")
+                    if param_type == "array":
+                        if not param:
+                            continue
+                        assert isinstance(param, Iterable) and not isinstance(param, str)
+                        if not param_spec.get("explode", False):
+                            param = ",".join(param)
+                    elif param_type == "integer":
+                        assert isinstance(param, int)
+                result[name] = param
```

Comment:
Proper handling of exploded or not exploded arrays kind of automatically leads to more validation of the parameters. ;)

Reply:
And extra validation may lead to more bugs uncovered... Need to find time investigate these...

Reply:
I put the validation and array substitution on a separate branch to investigate separately:

https://github.com/mdellweg/pulp-cli/tree/validate_parameters